plugins/git-release-manager/src/api/GitReleaseClient.ts

```diff
@@ -0,0 +1,837 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigApi, OAuthApi } from '@backstage/core';
+import { Octokit } from '@octokit/rest';
+import { ScmIntegrations } from '@backstage/integration';
+
+import { DISABLE_CACHE } from '../constants/constants';
+import { Project } from '../contexts/ProjectContext';
+import { UnboxArray, UnboxReturnedPromise } from '../types/helpers';
+import { GitReleaseManagerError } from '../errors/GitReleaseManagerError';
+
+export class GitReleaseClient implements GitReleaseApi {
+  private readonly githubAuthApi: OAuthApi;
+  private readonly apiBaseUrl: string;
+  readonly host: string;
+
+  constructor({
+    configApi,
+    githubAuthApi,
+  }: {
+    configApi: ConfigApi;
+    githubAuthApi: OAuthApi;
+  }) {
+    this.githubAuthApi = githubAuthApi;
+
+    const { host, apiBaseUrl } = this.getGithubIntegrationConfig({ configApi });
+
+    this.host = host;
+    this.apiBaseUrl = apiBaseUrl;
+  }
+
+  private getGithubIntegrationConfig({ configApi }: { configApi: ConfigApi }) {
+    const integrations = ScmIntegrations.fromConfig(configApi);
```

Comment:
could have passed this in as a dependency instead of the config, so it's easier to mock